### ISSUE TITLE
Jumpy cursor when launching 'nvim' inside tmux 2.5 under iTerm2 if highlight Normal is set.

### ISSUE THREAD

**Issue by bluz71** [neovim/neovim#7015]
Hello support,

`nvim` version 0.2.0.1 (Homebrew)
`tmux` version 2.5 (Homebrew)
`iTerm2` version 3.0.15
`macOS` version Sierra

My machine is a slow Macbook with Core-M processor; I assume the slower the machine the more noticeable the issue is.

This is the minimal *init.vim*:

```
highlight Normal ctermbg=232 ctermfg=251
```

When launching *nvim*, under tmux, the cursor jumps all over the screen before it settles in the top left corner. You must look very carefully since Neovim launches so quick. There is no slowdown or any other deleterious effect; just the cursor jumpiness as Neovim launches. I suggest launching and exiting *nvim* a few times to see this issue (slower machines the better).

My recollection is that this is new. Maybe when I upgraded to *tmux* 2.5; though I can't be certain of this since I use *nvim* and *Vim* interchangeably; maybe the jumpiness has been there for a while and I haven't noticed. Actually from memory I upgraded to Neovim 0.2.0.1 at the same time as tmux; small chance *0.2.0.1* could be a culprit (though very unlikely). No jumpiness with Vim 8.0.0703. No jumpiness when **NOT** using tmux.

Also, when the highlight `Normal` group line is removed the jumpiness goes away.

So to make sure I am not going mad I simply ask another Apple Mac Neovim user to see if they can reproduce this issue on their machine using the minimal *init.vim*. 

Lastly, I do **not** see this on my Linux machine  also running tmux 2.5 & Neovim 0.2.0.

Thanks.

**Comment by tbodt** [comment]
When the `highlight Normal` is present, every single cell in the terminal grid has to be redrawn with the background color, which requires moving the cursor to every cell and ouputting a space. This cursor movement is visible on any terminal, tmux makes it worse since it's an extra layer that has to process the big dump of escape sequences.

**Comment by bluz71** [comment]
Interesting, that does make sense. *iTerm2* is also quite slow, I'm sure making the issue more obvious yet still.

The only question I have is why does Vim (8.0.698) not have the same jumpy cursor? Maybe it is still visible, but faster or hidden? Same machine, same terminal, same tmux, *nvim* zappy/jumpy cursor, *Vim* appears not to have same jumpy cursor; yes, just tested many times, Vim either hides the jumpiness very well or does not have it all.

Ok, I've passed on my experiences; I don't think a bug exists here; just a behaviour difference to Vim. I suspect this should be closed. Thanks for the answer.

**Comment by justinmk** [comment]
Looks related to https://github.com/neovim/neovim/issues/4210 . Thanks for the extra information. I moved that issue from "unplanned" to "todo", it is clear now that it's a regression.

**Comment by ZyX-I** [comment]
@tbodt This should be something else: check `nvim -u <(echo 'sleep 5')`, it will show the UI only after 5 second delay, so at that point there is nothing to *redraw* because nothing was drawn so far. `nvim -u <(echo $'sleep 5\nhi Normal ctermfg=Red\nsleep 5')` also shows anything only after 10 second delay.

**Comment by ZyX-I** [comment]
I was testing with #6364 though, maybe something in master made things better or worse.

**Comment by bluz71** [comment]
@ZyX-I ,

I did `nvim -u <(echo 'sleep 5')` and there is **no** jumpy cursor (also for 1 second delay). 
Same for `nvim -u <(echo $'sleep 1\nhi Normal ctermbg=232 ctermfg=251\nsleep 1')`, **no** jumpy cursor.
However, `nvim -u <(echo $'hi Normal ctermbg=232 ctermfg=251')` does have jumpy cursor.

Emphasis on tmux being a key ingredient (possibly due to it slowing everything down).

Is that helpful to you?

@justinmk,

This *may* relate to #4210; though this one is less about screen flickering as compared to the cursor jumping all over the screen before settling in the top left corner (just for tmux + iTerm2). I don't think I have any flickering per-se.

I closed this yesterday as *not a bug* before the latest round of comments, clearly something is amiss here and/or #4210. Should I leave this closed or reopen?
